ravenframework/utils/randomUtils.py

```diff
@@ -168,12 +191,100 @@ def getRNGSeed(self):
     self._seed = abs(int(val))
     return self._seed
 
+# FIXME: A wrapper class for numpy.random.Generator is used here instead of using the Generator directly
+# in order to reproduce the same outputs as CrowRNG.random(). CrowRNG, which uses the 32-bit
+# boost::random::mt19937 under the hood in Crow, produces 64-bit floats by dividing a 32-bit integer
+# by the maximum value of an unsigned 32-bit integer; this division produces a 64-bit float. This is
+# far from ideal, since this does not actually contain 64 bits of randomness. However, fixing this
+# or using an RNG which does not produce identical outputs to CrowRNG.random() would require enormous
+# effort to fix all of the tests which rely on the current behavior. At the time of writing, that is
+# approximately 400 of about 900 tests! If anybody has the availability to fix that many tests, please
+# do it! Switching the util functions in this file to directly use the methods of np.random.Generator
+# really streamline things and probably provide a reasonable speed boost. But for now, we'll go with
+# the wrapper option so we don't have to regold all those tests.
+#   j-bryan, 2021-03-11
+class NumpyRNG:
+  """ Wraps numpy.random.Generator to provide an interface similar to CrowRNG """
+  def __init__(self):
+    """
+      Constructor
+      @ In, None
+      @ Out, None
+    """
+    self._seed = 5489  # default seed of boost::random::mt19937
+    bitGenerator = np.random.MT19937()
+    # MT19937(seed) doesn't produce the same initial seed and state as the _legacy_seeding method
+    # because passing the seed in the MT19937 constructor or the Generator.seed() method creates a
+    # SeedSequence object, which effectively hashes the seed value, while _legacy_seeding uses the
+    # seed value directly. FIXME: If somebody in the future knows of a way to get the same initial
+    # state from the seed without using a private method, please change this! -- j-bryan, 2023-11-09
+    bitGenerator._legacy_seeding(self._seed)
+    self._engine = np.random.Generator(bitGenerator)
+
+  def seed(self, value):
+    """
+      Reseeds the RNG
+      @ In, value, int, RNG seed
+      @ Out, None
+    """
+    self._seed = abs(int(value))
+    # According to the numpy docs, best practice is to create a new Generator rather than reseed an
+    # existing one.
+    bitGenerator = np.random.MT19937()
+    bitGenerator._legacy_seeding(self._seed)
+    self._engine = np.random.Generator(bitGenerator)
+
+  def getRNGSeed(self):
+    """
+      Gets the RNG seed
+      @ In, None
+      @ Out, int, RNG seed value
+    """
+    return self._seed
+
+  def random(self, size):
+    """
+      Uses a scaling of Generator.integers because Generator.random() does not return values in the same order
```

Comment:
Possibly this should mention that this only returns about 32 bits of randomness.